### PR TITLE
Upgrade PostgreSQL driver to 42.3.1

### PR DESCRIPTION
**Pull request by wendigo**
cc @hashhar

### CODE REVIEW

File: plugin/trino-postgresql/src/main/java/io/trino/plugin/postgresql/TypeUtils.java [hunk 1]
```diff
@@ -75,7 +75,7 @@ private TypeUtils() {}
     static String getArrayElementPgTypeName(ConnectorSession session, PostgreSqlClient client, Type elementType)
     {
         if (DOUBLE.equals(elementType)) {
-            return "float";
+            return "float8";
```
Comment (hashhar) [inline plugin/trino-postgresql/src/main/java/io/trino/plugin/postgresql/TypeUtils.java:78]: Good catch. Any idea why it didn't cause issues before?

File: plugin/trino-postgresql/src/main/java/io/trino/plugin/postgresql/PostgreSqlClient.java [hunk 2]
```diff
@@ -224,9 +224,7 @@
     private static final int ARRAY_RESULT_SET_VALUE_COLUMN = 2;
     private static final String DUPLICATE_TABLE_SQLSTATE = "42P07";
     private static final int POSTGRESQL_MAX_SUPPORTED_TIMESTAMP_PRECISION = 6;
-
-    // TODO: Since 42.2.23 the JDBC driver will return 0 as column size for decimal types with unspecified precision
-    private static final int PRECISION_OF_UNSPECIFIED_DECIMAL = 131089;
+    private static final int PRECISION_OF_UNSPECIFIED_DECIMAL = 0;
```
Comment (hashhar) [inline plugin/trino-postgresql/src/main/java/io/trino/plugin/postgresql/PostgreSqlClient.java:227]: I'd be so happy with myself if this is the only change needed to get DECIMALs to work correctly.

**Review by hashhar — APPROVED** [review]
LGTM % ci

File: plugin/trino-postgresql/src/main/java/io/trino/plugin/postgresql/TypeUtils.java (hunk 1 above)
Comment (wendigo) [reply plugin/trino-postgresql/src/main/java/io/trino/plugin/postgresql/TypeUtils.java:78]: I'm not sure, but it was wrong in the first place - I've examined what type is returned when reading `double` and it is `float8`.

**Comment by hashhar** [comment]
```
Error:  io.trino.plugin.iceberg.TestIcebergOrcConnectorTest.testAllAvailableTypes  Time elapsed: 3.827 s  <<< FAILURE!
java.lang.AssertionError: 
not equal
Actual rows (up to 100 of 1 extra rows shown, 18 rows in total):
    [a_boolean, null, 0.0, null, true, true]
Expected rows (up to 100 of 1 missing rows shown, 18 rows in total):
    [a_boolean, null, 0.0, null, null, null]

	at org.testng.Assert.fail(Assert.java:94)
	at io.trino.testing.QueryAssertions.assertEqualsIgnoreOrder(QueryAssertions.java:237)
	at io.trino.testing.QueryAssertions.assertEqualsIgnoreOrder(QueryAssertions.java:223)
	at io.trino.plugin.iceberg.BaseIcebergConnectorTest.lambda$testAllAvailableTypes$31(BaseIcebergConnectorTest.java:2568)
	at org.assertj.core.api.AbstractAssert.satisfies(AbstractAssert.java:813)
	at io.trino.plugin.iceberg.BaseIcebergConnectorTest.testAllAvailableTypes(BaseIcebergConnectorTest.java:2540)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:104)
	at org.testng.internal.Invoker.invokeMethod(Invoker.java:645)
	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:851)
	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:1177)
	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:129)
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:112)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:829)
```

cc: @findepi

**Comment by electrum** [comment]
Thanks @wendigo! @hashhar great work getting this fixed upstream!

**Pull request merged**